Update and slightly clean up `Privacy`

Userlog2 opened the pull request:
Corrected punctuation and fixed typos, and updated services (namely, Skype no longer exists and so cannot be added to user profiles).

I was also holding back on removing "replay comments" since those *technically* still exist.

- [x] remember to update "last updated" date before merging

Userlog2 committed:
Update and slightly clean up `Privacy`

Corrected punctuation and fixed typos, and updated services (namely, Skype no longer exists and so cannot be added to user profiles).

Userlog2 committed:
Update "last updated" date

Walavouchey commented on wiki/Legal/Privacy/en.md:
```diff
@@ -89,14 +89,14 @@ We may share your personal data with third parties in very specific cases:
-- To process your customer service tickets (we use [Enchant](https://enchant.com))
-- In order to improve our service, via error logging (we use [Sentry](https://sentry.io))
+- To process your customer service tickets, (we use [Enchant](https://enchant.com))
+- In order to improve our service via error logging, (we use [Sentry](https://sentry.io))
```
this one doesn't make any sense. previous was correct

Walavouchey reviewed (commented):
rest is good

Userlog2 committed:
walareviü

MChecaH commented on wiki/Legal/Privacy/en.md:
```diff
@@ -90,13 +90,13 @@ We may share your personal data with third parties in very specific cases:
+As a user, you have the right to migrate, update, or delete your personal information. This can be done primarily from the [user settings](https://osu.ppy.sh/home/account/edit), or where not available from a localised "Edit" feature on the relevant section of our site. In case you wish to programmatically retrieve your full account data, please use our [public API](https://github.com/ppy/osu-api/ ...
 
-In many cases, user submissions such as forum posts and beatmaps can be deleted on an individual basis. You will find delete buttons directly associated with the items that can be deleted.
+In many cases, user submissions, such as forum posts and beatmaps, can be deleted on an individual basis. You will find delete buttons directly associated with the items that can be deleted.
```
Feels better to use the dash to avoid repeating the comma, yeah. Will just need to be em dashes rather than en.

```suggestion
In many cases, user submissions — such as forum posts and beatmaps — can be deleted on an individual basis. You will find delete buttons directly associated with the items that can be deleted.
```

I dislike this language's standards over the length of a dash...

Userlog2 committed:
Add MChecaH's suggested em dashes

Co-authored-by: MChecaH <93517213+MChecaH@users.noreply.github.com>

peppy committed:
Update disclosures section

- We self-host sentry
- All merch is handled externally now, we stopped doing fulfilment processing on our side. All user data has been deleted.
- Clarified our stance on analytics

peppy committed:
Update last updated and fix some grammar

Userlog2 committed:
one final fix

Userlog2 committed:
Walareviü

Co-authored-by: Walavouchey <36758269+Walavouchey@users.noreply.github.com>

peppy committed:
Update date
